### ISSUE TITLE
salt.states.file.copy does not copy file and fails with "Group root is not available"

### ISSUE THREAD

**Issue by nmadhok** [saltstack/salt#24648]
Here's the error i get when I run the state:

```
----------
          ID: Revert to original /etc/passwd
    Function: file.copy
        Name: /etc/passwd
      Result: False
     Comment: Group root is not available
     Started: 17:19:51.808226
    Duration: 7.653 ms
     Changes:   
----------
          ID: Revert to original /etc/shadow
    Function: file.copy
        Name: /etc/shadow
      Result: False
     Comment: Group root is not available
     Started: 17:19:51.818615
    Duration: 7.565 ms
     Changes:   
```

Here's the state I'm using:

```
Revert to original /etc/passwd:
  file.copy:
    - name: /etc/passwd
    - source: /etc/passwd.orig
    - force: True
    - unless: diff /etc/passwd /etc/passwd.orig

Revert to original /etc/shadow:
  file.copy:
    - name: /etc/shadow
    - source: /etc/shadow.orig
    - force: True
    - unless: diff /etc/shadow /etc/shadow.orig
```

This also fails on salt-enterprise 4.0.2:

```
           Salt: 4.0.2
         Python: 2.6.6 (r266:84292, Jan 22 2014, 01:49:05)
         Jinja2: 2.2.1
       M2Crypto: 0.20.2
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.5.0
           RAET: Not Installed
            ZMQ: 4.0.5
           Mako: Not Installed
```

**Comment by jfindlay** [comment]
@nmadhok, thanks for the report.

**Comment by tedski** [comment]
I'm having trouble reproducing this, @nmadhok.  Can you provide for information about the OS and what the host uses for group resolution (ldap or files or other?).

**Comment by leeclemens** [comment]
TL;DR: Considering lack of OS/platform information, this issue might be able to be closed.

I ran into this same error while using FreeBSD 10.

State file:
```
/etc/blahblah:
  file:
    - managed
    - source: salt://files/blahblah
    - user: root
    - group: root
    - mode: '0644'
```
Result:
```
Comment: Group root is not available
```

`$ grep root /etc/group`
```
wheel:*:0:root
operator:*:5:root
```

```
Salt Version:
           Salt: 2015.8.0
```

Not sure if the same cause as OP ran into, but in FreeBSD 10 there is no group `root`. On CentOS 6, `root` is a member of the `wheel` group, but the group ID of `wheel` is 10...not 0 (as would be expected from a salt state config using `group: root`.

Changing the state file to use `user: 0` and `group: 0` works on both platforms.

**Comment by leeclemens** [comment]
Related to FreeBSD not having a group named `root`? (I encountered the same error, `Group root is not available` and got here via Google.)  Stab in the dark...seems like info requested by @tedski is necessary to understand what the issue may be.
